pages/docs/couple-your-code/couple-your-code-gradient-data.md

```diff
@@ -0,0 +1,161 @@
+* `ẁriteBlockVectorGradintData` writes multiple vector gradient data at once (analogue to `writeBlockVectorData`)
+
+Let's consider an example for writing block vector gradient data corresponding to the vector data `v0 = (v0x, v0y) , v1 = (v1x, v1y), ... , vn = (vnx, vny)` differentiated in spatial directions x and y.
+Per default, the values are passed as following:
+
+```cpp
+( v0x_dx, v0x_dy, v0y_dx, v0y_dy,
+  v1x_dx, v1x_dy, v1y_dx, v1y_dy,
+  ... ,
+  vnx_dx, vnx_dy, vny_dx, vny_dy  )  
+```
+
+When using the `rowsFirst` parameter, the following format is required:
+
+```cpp
+( v0x_dx, v0y_dx, v1x_dx, v1y_dx, ... , vnx_dx, vny_dx,
+  v0x_dy, v0y_dy, v1x_dy, v1y_dy, ... , vnx_dy, vny_dy  )
+```
+
+Let's add gradient data to our example code:
+
+```cpp
+turnOnSolver(); //e.g. setup and partition mesh
+
+precice::SolverInterface precice("FluidSolver","precice-config.xml",rank,size); // constructor
+
+// set mesh elements
+int dim = precice.getDimensions();
+int meshID = precice.getMeshID("FluidMesh");
+int vertexSize; // number of vertices at wet surface
+
+// determine vertexSize
+double* coords = new double[vertexSize*dim]; // coords of coupling vertices
+// determine coordinates
+int* vertexIDs = new int[vertexSize];
+precice.setMeshVertices(meshID, vertexSize, coords, vertexIDs);
+delete[] coords;
+
+// create data first
+int displID = precice.getDataID("Displacements", meshID);
+int stressID = precice.getDataID("Stress", meshID);
+
+double* Stress = new double[vertexSize * dim];
+double* displacements = new double[vertexSize * dim];
+
+// create gradient data
+double* stressGradient = new double[vertexSize * dim * dim]
+
+[...]
+precice_dt = precice.initialize();
+
+while (not simulationDone()){ // time loop
+  precice.readBlockVectorData(displID, vertexSize, vertexIDs, displacements);
+  setDisplacements(displacements);
+  [...]
+  solveTimeStep(dt);
+  computeStress(Stress);
+
+  precice.writeBlockVectorData(stressID, vertexSize, vertexIDs, Stress);
+
+  // write gradient data
+  if (isGradientDataRequired(dataID)){
+    computeStressGradient(stressGradient)
+    precice.writeBlockVectorGradientData(stressID, vertexSize, vertexIDs, stressGradient);
+  }
+
+  precice_dt = precice.advance(dt);
+}
+[...]
+```
+
+{% experimental %}
+This is an experimental feature.
+{% endexperimental %}
+
+As a last step, you need to set the flag `gradient="on"` in the configuration file, whenever you require to write gradient data.
+
+For the example, you can use the following `precice-config.xml`:
+
+```xml
+<?xml version="1.0"?>
+
+<precice-configuration>
+
+  <solver-interface dimensions="3" experimental="on">
+
+    <data:vector name="Stress" gradient="on"/>
+    <data:vector name="Displacements" gradient="on" />
```

Comment:
```suggestion
    <data:vector name="Displacements" />
```

Above we only use gradients for displacements. I think such a mixed example could be good anyway.